src/main/java/appeng/items/storage/ItemReadOnlyStorageCell.java

```diff
@@ -0,0 +1,240 @@
+		this.totalBytes = 64 * 1024;
+		this.idleDrain = 2.0;
+		this.perType = 512;
+	}
+
+	@Override
+	public void addCheckedInformation( final ItemStack stack, final EntityPlayer player, final List<String> lines, final boolean displayMoreInfo )
+	{
+		final IMEInventoryHandler<?> inventory = AEApi.instance().registries().cell().getCellInventory( stack, null, StorageChannel.ITEMS );
+
+		if( inventory instanceof ICellInventoryHandler )
+		{
+			final ICellInventoryHandler handler = (ICellInventoryHandler) inventory;
+			final ICellInventory cellInventory = handler.getCellInv();
+
+			if( cellInventory != null )
+			{
+				if( getReadOnly( stack ) )
+				{
+					if( cellInventory.getUsedBytes() == 0 )
+					{
+						lines.add( GuiText.Empty.getLocal() );
+					}
+					else
+					{
+						lines.add( cellInventory.getUsedBytes() + " " + GuiText.BytesUsed.getLocal() );
+
+						lines.add( cellInventory.getStoredItemTypes() + " " + GuiText.Types.getLocal() );
+					}
+				}
+				else
+				{
+					lines.add( cellInventory.getUsedBytes() + " " + GuiText.Of.getLocal() + ' ' + cellInventory.getTotalBytes() + ' ' + GuiText.BytesUsed.getLocal() );
+
+					lines.add( cellInventory.getStoredItemTypes() + " " + GuiText.Of.getLocal() + ' ' + cellInventory.getTotalItemTypes() + ' ' + GuiText.Types.getLocal() );
+				}
+			}
+		}
+	}
+
+	@Override
+	public int getBytes( final ItemStack cellItem )
+	{
+		return this.totalBytes;
+	}
+
+	@Override
+	public int getBytesPerType( final ItemStack cellItem )
+	{
+		return this.perType;
+	}
+
+	@Override
+	public int getTotalTypes( final ItemStack cellItem )
+	{
+		return 63;
+	}
+
+	@Override
+	public boolean isBlackListed( final ItemStack cellItem, final IAEItemStack requestedAddition )
+	{
+		return getReadOnly( cellItem );
+	}
+
+	@Override
+	public boolean storableInStorageCell()
+	{
+		return false;
+	}
+
+	@Override
+	public boolean isStorageCell( final ItemStack i )
+	{
+		return true;
+	}
+
+	@Override
+	public double getIdleDrain()
+	{
+		return this.idleDrain;
+	}
+
+	@Override
+	public String getUnlocalizedGroupName( final Set<ItemStack> others, final ItemStack is )
+	{
+		return GuiText.StorageCells.getUnlocalized();
+	}
+
+	@Override
+	public boolean isEditable( final ItemStack is )
+	{
+		return false;
+	}
+
+	@Override
+	public IInventory getUpgradesInventory( final ItemStack is )
+	{
+		return new CellUpgrades( is, 0 );
+	}
+
+	@Override
+	public IInventory getConfigInventory( final ItemStack is )
+	{
+		return new CellConfig( is );
+	}
+
+	@Override
+	public FuzzyMode getFuzzyMode( final ItemStack is )
+	{
+		return FuzzyMode.IGNORE_ALL;
+	}
+
+	@Override
+	public void setFuzzyMode( final ItemStack is, final FuzzyMode fzMode )
+	{
+
+	}
+
+	@Override
+	public ActionResult<ItemStack> onItemRightClick( final ItemStack stack, final World world, final EntityPlayer player, final EnumHand hand )
+	{
+		this.lockDrive( stack, player );
+		return new ActionResult<ItemStack>( EnumActionResult.SUCCESS, stack );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2293)